### ISSUE TITLE
FTP_SERVER ARGS not defined error is coming

### ISSUE THREAD

**Issue by vedikinsolutions** [SamKirkland/FTP-Deploy-Action#36]
Hello!

I am trying to configure this git action in my repository. Here is my main.yml file looks like.

```
name: Publish Website
on: 
  push:
    branches:
    - master
jobs:
  deploy:
    name: FTP-Deploy-Action
    runs-on: ubuntu-latest
    steps:
      - name: FTP-Deploy-Action
        uses: SamKirkland/FTP-Deploy-Action@2.0.0
        with:
          FTP_SERVER: thcitsolutions.com
          ftp_username: dev@thcitsolutions.com
          ftp_password: ##ActualPasswordGoesHere##
          method: ftp
          port: 21
          # The local folder to copy, defaults to root project folder
          # local_dir: # optional, default is 
          # The remote folder to copy to, deafults to root FTP folder (I recommend you configure this on your server side instead of here)
          remote_dir: test # optional, default is 
          # Passes through options into lftp
          ARGS: --exclude=.ftpquota --exclude-glob=.git*/** --exclude-glob=.git**
```

But when I push to master branch, it is giving error while connecting to FTP (/entrypoint.sh: line 25: FTP_SERVER: parameter not set). PFA screenshot for reference purpose.

Can you please guide what I am missing. 
![GitHub FTP - Error Screenshot](https://user-images.githubusercontent.com/53016481/71943582-1eabbf80-31e7-11ea-8268-6976ff0bddd3.PNG)

Note: I am new to Git Action and this is my first git action.

**Comment by SamKirkland** [comment]
Hey, congratulations on using your first action!

Simple fix, your config is SO close.
Simply change `with:` -> `env:`

```yaml
name: Publish Website
on: 
  push:
    branches:
    - master
jobs:
  deploy:
    name: FTP-Deploy-Action
    runs-on: ubuntu-latest
    steps:
      - name: FTP-Deploy-Action
        uses: SamKirkland/FTP-Deploy-Action@2.0.0
        env:
          FTP_SERVER: thcitsolutions.com
          ftp_username: #########@thcitsolutions.com
          ftp_password: ##ActualPasswordGoesHere##
          method: ftp
          port: 21
          # The local folder to copy, defaults to root project folder
          # local_dir: # optional, default is 
          # The remote folder to copy to, deafults to root FTP folder (I recommend you configure this on your server side instead of here)
          remote_dir: test # optional, default is 
          # Passes through options into lftp
          ARGS: --exclude=.ftpquota --exclude-glob=.git*/** --exclude-glob=.git**
```

Please post back and let me know if that resolved your issue, then we can continue debugging or close the issue.

**Comment by vedikinsolutions** [comment]
@SamKirkland Thank you so much for such a quick response. Now FTP task is showing successful. Remote directory provided in config is created in FTP successfully. However when I create new file or update any existing files in master branch, they are not getting uploaded in FTP. 

Remote directory is created through git action. Hence it is confirm that it is able to connect to FTP successfully. But files are not getting uploaded yet. 

Am I still missing anything?

**Comment by SamKirkland** [comment]
Hmmm not sure. Can you add the `--verbose` flag to your `ARGS:` and see post your logs here?

**Comment by SamKirkland** [comment]
@vedikinsolutions Are you still having problems?
If not i'll close out this issue.

**Comment by ryaneastabrook** [comment]
If you're inline editing your workflow file on Github, you'll see an example which has "with" instead of "env". I'm sure this is where the confusion is coming from. Not sure how to change or I'd submit a PR.

![image](https://user-images.githubusercontent.com/595337/74448750-559d7280-4e30-11ea-995f-6d58f3130db1.png)

**Comment by SamKirkland** [comment]
@ryaneastabrook good find!
It's been fixed but that document is wrong in the v2.0.0 release! I'll make another release so it uses the latest readme :)
